src/IO/UseSSL.h

```diff
@@ -1,12 +1,26 @@
 #pragma once
 
+#include "config.h"
+
 #include <boost/noncopyable.hpp>
+#include <atomic>
+
+#if USE_SSL
+#    include <openssl/provider.h>
+#endif
 
 namespace DB
 {
 // http://stackoverflow.com/questions/18315472/https-request-in-c-using-poco
 struct UseSSL : private boost::noncopyable
 {
+    std::atomic<uint8_t> ref_count{0};
```

Comment:
I guess we should rename this file and the class name to UseOpenSSL.h (just "SSL" is misleading).

Reply:
It will be a new `Common/Crypto/OpenSSLInitializer.h`, just didn't get to final polishing yet.